poppy/instrument.py

```diff
@@ -886,38 +893,54 @@ def _get_weights(self, source=None, nlambda=None, monochromatic=None, verbose=Fa
 
             poppy_core._log.info("Computing wavelength weights using synthetic photometry for %s..." % self.filter)
             band = self._get_synphot_bandpass(self.filter)
+            band_wave = band.waveset
+            band_thru = band(band_wave)
+
+            # Update source to ensure that it covers the entire filter
+            if band_wave.value.min() < source.waveset.value.min() or \
+                    band_wave.value.max() > source.waveset.value.max():
+                source_meta = source.meta
+                wave, wave_str = synphot.utils.generate_wavelengths(band_wave.value.min(), band_wave.value.max(),
+                                                                    wave_unit=units.angstrom, log=False)
+                source = synphot.SourceSpectrum(Empirical1D, points=wave, lookup_table=source(wave))
+                source.meta.update(source_meta)
+
             # choose reasonable min and max wavelengths
-            w_above10 = np.where(band.throughput > 0.10 * band.throughput.max())
+            w_above10 = np.where(band_thru > 0.10 * band_thru.max())
 
-            minwave = band.wave[w_above10].min()
-            maxwave = band.wave[w_above10].max()
-            poppy_core._log.debug("Min, max wavelengths = %f, %f" % (minwave / 1e4, maxwave / 1e4))
+            minwave = band_wave[w_above10].min()
+            maxwave = band_wave[w_above10].max()
+            poppy_core._log.debug("Min, max wavelengths = %f, %f" % (
+                minwave.to_value(units.micron), maxwave.to_value(units.micron)))
 
             wave_bin_edges = np.linspace(minwave, maxwave, nlambda + 1)
             wavesteps = (wave_bin_edges[:-1] + wave_bin_edges[1:]) / 2
             deltawave = wave_bin_edges[1] - wave_bin_edges[0]
+            jwst_area = 25.4 * (units.m * units.m)
```

Comment:
We can simply delete this line with the jwst_area and remove the usage of it below. poppy is not interested in absolute fluxes, just normalized PSFs, so we can simplify and avoid hard-coding in this parameter with no change in results.

Reply:
So `area` is a required parameter for `synphot.Observation`, but we can change it so it just takes in 1 m^2 rather than a specific number related to JWST.

Reply:
`area` is required but it is up to you what value to give it.

In PySynphot, it was also required but PySynphot just automatically used HST area unless you overwrite the setting.